lib/Myriad/Config.pm

```diff
@@ -135,11 +129,131 @@ method define ($key, $v) {
     $config->{$key} = $DEFAULTS{$key} = Ryu::Observable->new($v);
 }
 
+method parse_subargs ($subarg, $root, $value) {
+    $subarg =~ s/(.*)[_|\.](configs?|instances?)(.*)/$2$3/;
+    die 'invalid service name' unless $2;
+
+    my $service_name = $1;
+    $service_name =~ s/_/\./g;
+    $root = $root->{$service_name} //= {};
+
+    my @sublist = split /_|\./, $subarg;
+    die 'config key is not formated correctly' unless @sublist;
+    while (@sublist > 1) {
+        my $level = shift @sublist;
+        $root->{$level} //= {};
+        $root= $root->{$level};
+    }
+    $root->{$sublist[0]} = $value;
+}
+
+method from_args ($commandline) {
```

Comment:
Moving these out is good, but they need documentation and the names aren't really following the right convention: each of these *does* something, so it needs a verb. `->from_args` implies that it's returning data - `my $something = $obj->from_something_else`.